default/python/AI/AIstate.py

```diff
@@ -158,12 +165,15 @@ def __setstate__(self, state):
                 convert_to_version(state, v+1)
             self.__dict__ = state
         except ConversionError:
-            try:
-                aggression = state['character'].get_trait(Aggression).key
-            except Exception as e:
-                print >> sys.stderr, "Could not find the aggression level of the AI, defaulting to typical."
-                print >> sys.stderr, e
-                aggression = fo.aggression.typical
+            if '_aggression' in state:
+                aggression = state['_aggression']
+            else:
+                try:
```

Comment:
I don't read total thread, but this looks strange for me.  We have handler for difference and have some separate code to handle difference.